addons/payment/models/payment_transaction.py

```diff
@@ -157,6 +161,7 @@ def create(self, values_list):
             acquirer = self.env['payment.acquirer'].browse(values['acquirer_id'])
 
             if not values.get('reference'):
+                # TODO ANVFE Does it happen to create transactions without reference ???
```

Comment:
TODO remove: Yes it happens when paying a specific SO (/my/orders/<id>/transaction iirc)